### PR TITLE
Added support to provisioning rke2 azure node driver clusters

### PR DESCRIPTION
**Pull request by igomez06**
**Framework updates** 
- Helper for creating a cloud credential for azure 
- Helper for creating a machine config for an azure provider.
- Updates to spelling mistakes in comments in the other cloud credential helpers

**Validation updates**
- Added azure to providers
- Updates to method names to adhere to go lang naming conventions

### CODE REVIEW

**Review by paynejacob — APPROVED** [review]

File: tests/framework/extensions/machinepools/azure_machine_config.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+)
+
+const (
+	AzureKind                              = "AzureConfig"
+	AzurePoolType                          = "rke-machine-config.cattle.io.azureconfig"
+	AzureResourceConfig                    = "azureconfigs"
+	AzureMachingConfigConfigurationFileKey = "azureMachineConfig"
+)
+
+type AzureMachineConfig struct {
```
Comment (izaac) [inline tests/framework/extensions/machinepools/azure_machine_config.go:15]: Are we sure these are all the possible parameters the config could receive?
Nit: add comment to the specific upstream configuration

**Review by izaac — CHANGES_REQUESTED** [review]
Looks good so far but I'd prefer to have the `AzureMachineConfig` (and potentially any other config object?) linked to their documentation in case we need a quick reference on what or how to format the values of each object.

File: tests/framework/extensions/machinepools/azure_machine_config.go (hunk 1 above)
Comment (igomez06) [reply tests/framework/extensions/machinepools/azure_machine_config.go:15]: Yes these are all the parameters. Is there something I am possibly missing Isaac? I just checked again on rancher, and this is it, unless I'm not seeing something. Other question what do you by "a comment to the specific upstream configuration" like a windows vs linux? Would be better in the actual test?

Comment (izaac) [reply tests/framework/extensions/machinepools/azure_machine_config.go:15]: Oh I thought this was coming from an upstream library but is it the Rancher API? In that case we're good.

Comment (igomez06) [reply tests/framework/extensions/machinepools/azure_machine_config.go:15]: Yes this is in Rancher API, ok sounds good thanks for checking.

**Review by izaac — APPROVED** [review]
LGTM

**Pull request merged**